Gems/Atom/RHI/Code/Include/Atom/RHI/MemorySubAllocator.h

```diff
@@ -7,11 +7,14 @@
  */
 #pragma once
 
+#include <AzCore/Debug/Profiler.h>
 #include <Atom/RHI/FreeListAllocator.h>
 #include <Atom/RHI/PoolAllocator.h>
 #include <Atom/RHI/MemoryAllocation.h>
 #include <Atom/RHI.Reflect/MemoryEnums.h>
 
+AZ_DECLARE_BUDGET(RHI);
+
```

Comment:
Is this needed? My understanding was that the budget just had to be declared once over the whole codebase.

Reply:
it has to be defined once, but you still need to have that symbol visible.
Under the hood the macro uses a function to get that budget

`AZ_DECLARE_BUDGET` -> declares the function
`AZ_DEFINE_BUDGET` -> defines the function
`AZ_PROFILE_FUNCTION` -> calls the function, so it needs to be able to see the declaration